src/stream/src/executor/backfill/cdc/cdc_backfill.rs

```diff
@@ -219,8 +222,13 @@ impl<S: StateStore> CdcBackfillExecutor<S> {
                 let left_upstream = upstream.by_ref().map(Either::Left);
 
                 let args = SnapshotReadArgs::new_for_cdc(current_pk_pos.clone(), self.chunk_size);
-                let right_snapshot =
-                    pin!(upstream_table_reader.snapshot_read(args).map(Either::Right));
+
+                #[auto_enum(futures03::Stream)]
+                let right_snapshot = if !paused {
+                    upstream_table_reader.snapshot_read(args).map(Either::Right)
+                } else {
+                    futures::stream::pending()
+                };
```

Comment:
What about pausing the `upstream` side as well? Does that need to be done?

Reply:
This only pauses the snapshot stream

Reply:
Since I guess upstream side will also handle it.

Reply:
It's eventually from a source executor, and it's already handled.

Reply:
Oh that will be paused, since `StreamSource` will be on that side, which supports `is_pause_on_start_up` already.

```
    StreamMaterialize { columns: [v1, v2], stream_key: [v1], pk_columns: [v1], pk_conflict: Overwrite }
    └─StreamUnion { all: true }
      ├─StreamExchange { dist: HashShard(mydb.t1.v1) }
      │ └─StreamCdcTableScan { table: mydb.t1, columns: [v1, v2] }
      └─StreamExchange { dist: HashShard(v1) }
        └─StreamDml { columns: [v1, v2] }
          └─StreamSource
```